test/Npgsql.Tests/ReaderTests.cs

```diff
@@ -1724,6 +1724,42 @@ await pgMock
             Assert.That(conn.FullState, Is.EqualTo(ConnectionState.Broken));
         }
 
+        [Test, Description("Timeouts ReadAsGetFieldValueAsync")]
+        [Timeout(15000)]
+        public async Task GetFieldValue_timeout_hard()
```

Comment:
For test completeness, we can also test IsDBNullAsync by passing a parameter in and branching on it (though not very important).

Possibly more important is to test cancellation in addition to timeout.